modules/porous_flow/include/materials/PorousFlowMaterialPermeabilityKozenyCarman.h

```diff
@@ -0,0 +1,92 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWMATERIALPERMEABILITYKOZENYCARMAN_H
+#define POROUSFLOWMATERIALPERMEABILITYKOZENYCARMAN_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowMaterialPermeabilityKozenyCarman;
+
+template<>
+InputParameters validParams<PorousFlowMaterialPermeabilityKozenyCarman>();
+
+/**
+ * Material designed to provide the permeability tensor which is calculated 
+ * from porosity using a form of the Kozeny-Carman equation (e.g. Oelkers 
+ * 1996: Reviews in Mineralogy v. 34, p. 131-192):
+ * k = k_ijk * A * phi^n / (1 - phi)^m
+ * where k_ijk is a tensor providing the anisotropy, phi is porosity, 
+ * n and m are positive scalar constants and A is given in one of the
+ * following forms:
+ * A = k0 * (1 - phi0)^m / phi0^n
+ * where k0 and phi0 are a reference permeability and porosity,
+ * or
+ * A = f * d^2
+ * where f is a scalar constant and d is grain diameter.
+ */
+class PorousFlowMaterialPermeabilityKozenyCarman : public DerivativeMaterialInterface<Material>
```

Comment:
If you derive from PorousFlowPermeabilityConst you won't need _k_anisotropy, _permeability, or _dpermeability_dvar.